pkg/resourcemanager/cosmosdb/sqldatabase/cosmosdbsql.go

```diff
@@ -0,0 +1,275 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package sqldatabase
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/services/cosmos-db/mgmt/2021-03-15/documentdb"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/Azure/azure-service-operator/api/v1alpha1"
+	"github.com/Azure/azure-service-operator/pkg/errhelp"
+	"github.com/Azure/azure-service-operator/pkg/helpers"
+	"github.com/Azure/azure-service-operator/pkg/resourcemanager"
+	"github.com/Azure/azure-service-operator/pkg/resourcemanager/config"
+	"github.com/Azure/azure-service-operator/pkg/resourcemanager/cosmosdb"
+	"github.com/Azure/azure-service-operator/pkg/resourcemanager/pollclient"
+)
+
+func NewAzureCosmosDBSQLDatabaseManager(creds config.Credentials) *AzureCosmosDBSQLDatabaseManager {
+	return &AzureCosmosDBSQLDatabaseManager{
+		Creds: creds,
+	}
+}
+
+var _ resourcemanager.ARMClient = &AzureCosmosDBSQLDatabaseManager{}
+
+type AzureCosmosDBSQLDatabaseManager struct {
+	Creds config.Credentials
+}
+
+/*
+ * ARMClient implementation
+ */
+
+func (m *AzureCosmosDBSQLDatabaseManager) Ensure(ctx context.Context, obj runtime.Object, opts ...resourcemanager.ConfigOption) (bool, error) {
+	options := &resourcemanager.Options{}
+	for _, opt := range opts {
+		opt(options)
+	}
```

Comment:
`options` isn't used in this method - is it just for side-effects in the ConfigOption callbacks?